Doc/tutorial/controlflow.rst

```diff
@@ -66,20 +66,20 @@ they appear in the sequence.  For example (no pun intended):
    window 6
    defenestrate 12
 
-If you need to modify the sequence you are iterating over while inside the loop
-(for example to duplicate selected items), it is recommended that you first
-make a copy.  Iterating over a sequence does not implicitly make a copy.  The
-slice notation makes this especially convenient::
-
-   >>> for w in words[:]:  # Loop over a slice copy of the entire list.
-   ...     if len(w) > 6:
-   ...         words.insert(0, w)
-   ...
-   >>> words
-   ['defenestrate', 'cat', 'window', 'defenestrate']
-
-With ``for w in words:``, the example would attempt to create an infinite list,
-inserting ``defenestrate`` over and over again.
+Code that modifies a collection while iterating over that same collection can
+be tricky to get right.  Instead, it is usually more straight-forward to loop
+over a copy of the collection or to create a new collection::
+
+    # Strategy:  Iterate over a copy
+    for user, status in users.copy().items():
+        if status == 'inactive':
+            del users[user]
+
+    # Strategy:  Create a new collection
+    active_users = {}
+    for user, status in users.items():
+        if status == 'active':
+            active_users[user] = status
```

Comment:
This is correct yet not idiomatic.
`active_users = {user:status for user, status in users.items if status == 'active'}`
For the previous example,
`long_words = [w for w in words if len(w) > 6]`
I presume comprehensions have not yet been discussed, but if they are later, I would add a forward reference.  Perhaps "See xxx for how to do this more compactly."

Reply:
* Comprehensions haven't been discussed yet. 
* Also, they are a distraction from the key point, "don't mutate while iterating"
* An unrolled loop is still considered valid, correct Python.  Comprehensions are not required.
* I prefer to keep this example as-is.  It is the simplest example of filtering an existing structure into a new structure rather than deleting entries.
* IMO, forward references are distracting and make the tutorial harder to read.
* At the point in the tutorial, we can't drown a person.  This section is the *first* introduction to a for-loop.